Enable static build

Already running fine in Linux/OSX on Travis, Windows is having issues:

```
aeron_driver_static.lib(aeron_fileutil.obj) : error LNK2019: unresolved external symbol __imp_SHFileOperationA referenced in function aeron_delete_directory
aeron_driver_static.lib(aeron_socket.obj) : error LNK2019: unresolved external symbol GetAdaptersAddresses referenced in function getifaddrs
aeron_driver_static.lib(aeron_socket.obj) : error LNK2019: unresolved external symbol ConvertLengthToIpv4Mask referenced in function getifaddrs
aeron_driver_static.lib(aeron_netutil.obj) : error LNK2019: unresolved external symbol if_nametoindex referenced in function aeron_ip_lookup_func
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -18,7 +18,7 @@ script:
   - cd aeron_driver-sys
   - cargo test
   - cargo test --release
-  #- cargo test --features "static"
+  - cargo test --features "static"
 
   - cd ..
   - cargo test

--- a/aeron_driver-sys/build.rs
+++ b/aeron_driver-sys/build.rs
@@ -46,6 +46,19 @@ pub fn main() {
         link_type.link_lib(),
         link_type.target_name()
     );
+
+    match link_type {
+        LinkType::Static => {
+            // On Windows, there are some extra libraries needed for static link
+            // that aren't included by Aeron.
+            if cfg!(target_os = "windows") {
+                println!("cargo:rustc-link-lib=shell32");
+                println!("cargo:rustc-link-lib=iphlpapi");
+            }
+        },
+        _ => ()
+    };
+
     let cmake_output = Config::new(&aeron_path)
         .build_target(link_type.target_name())
         .build();

--- a/aeron_driver-sys/build.rs
+++ b/aeron_driver-sys/build.rs
@@ -55,8 +55,8 @@ pub fn main() {
                 println!("cargo:rustc-link-lib=shell32");
                 println!("cargo:rustc-link-lib=iphlpapi");
             }
-        },
-        _ => ()
+        }
+        _ => (),
     };
 
     let cmake_output = Config::new(&aeron_path)

